File: _posts/Programmers-main/프로그래머스/unrated/기사단원의 무기/기사단원의 무기.py
from math import sqrt, trunc

def solution(number, limit, power):
    answer = 0
    for i in range(1, number+1):
        temp = 0
        if i == 1:
            temp += 1
        elif i % 2 == 0:
            for j in range(1, trunc(sqrt(i))+1):
                if i % j == 0:
                    if j ** 2 == i:
                        temp += 1
                    else:
                        temp += 2
        else:
            for j in range(1, trunc(sqrt(i))+2):
                if i % j == 0:
                    if j ** 2 == i:
                        temp += 1
                    else:
                        temp += 2 
        if temp > limit:
            temp = power
        answer += temp
    return answer

File: _posts/Programmers_cpp-main/프로그래머스/unrated/기사단원의 무기/기사단원의 무기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int number, int limit, int power) {
    int answer = 0;
    vector<int> list;
    for(int i = 1; i <= number; i++)
    {
        if (i == 1) list.push_back(1);
        else
        {
            int temp = 0;
            for (int j = 1; j * j <= i; j++)
            {
                if (j * j == i) temp += 1;
                else if (i % j == 0) temp += 2;
            }
            list.push_back(temp);
        }
    }
    for (int k = 0; k < list.size(); k++)
    {
        if (list[k] > limit) answer += power;
        else answer += list[k];
    }
    return answer;
}